ExploreASL_Master.m

```diff
@@ -71,23 +71,33 @@
     
     % -----------------------------------------------------------------------------
     % Import Master
-    x = ExploreASL_ImportMaster(x);
+    if x.opts.bImportData
+        x = ExploreASL_ImportMaster(x);
+    end
+    % Store logging information about errors/warnings in backup variable
+    if isfield(x,'logging')
+    	loggingBackUp = x.logging;
+    end
     
     % -----------------------------------------------------------------------------
     % Re-Initialize for potential data loading/processing
-    if x.bReinitialize > 0
-        x = ExploreASL_Initialize(x.DataParPath, x.ImportModules, x.ProcessModules, x.bPause, x.iWorker, x.nWorkers);
+    if x.opts.bReinitialize > 0
+        x = ExploreASL_Initialize(x.opts.StudyRoot, x.opts.ImportModules, x.opts.ProcessModules, x.opts.bPause, x.opts.iWorker, x.opts.nWorkers);
+    end
+    % Retrieve logging information about errors/warnings from backup variable
+    if exist('loggingBackUp', 'var')
+    	x.logging = loggingBackUp;
     end
     
     % -----------------------------------------------------------------------------
     % Print user feedback
-    if x.bProcessData==0 || x.bProcessData==2
-        if x.bProcessData==2 && nargout==0
+    if x.opts.bProcessData==0 || x.opts.bProcessData==2

```

Comment:
This is now confusing, we have 4 import "modules" and 3 processing "modules" but we still have this boolean (instead of the [0 0 1] [1 0 1] etc) so we should probably homogenize this when we split a common starting script from ImportMaster and ProcessingMaster

Reply:
No, please let me clarify this. For the import we have the [x x x x] array. If a single value is greater than 0, I set bImportData to 1. For the processing we have the [x x x] array. If a single value is greater than 0, I set bProcessData to 1. This is helpful for code readability, because otherwise we have a lot of statements with if sum([...])>0 etc. I don't like the possibility of bProcessData==2 either, but that's still a relic that we previously used if you only want to "load a dataset". It's exactly the behavior that you implemented a while ago, the only thing is that the variable name changed.

Reply:
I endorse this ;)

Reply:
True, it should then just become clear from the comments and header. Also, the bProcessData==2 is not necessary anymore, now this is the same as `ExploreASL_Master('/path2folder', 0,0)` right?

Reply:
weird, no it works.... I answered in a comment below, copy from there: bProcessData is no ExploreASL_Master input parameter anymore, but we still use it to determine if we want to load the data or not. Maybe there is another nice way to do this, but it was like a nicely running system before, so we kept it.